### PR TITLE
Add documentation for the manage membership team permission

### PR DESCRIPTION
**Pull request by JarrettSpiker**
### What
Adds documentation for the new Manage Membership team permission

[JIRA](https://hashicorp.atlassian.net/browse/TF-3937)

### Why
This permission allows org owners to delegate adding/removing users from teams, and from the org itself, to a different trusted team.

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] (N/A) Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] (N/A) Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] (N/A)New pages have metadata (page name and description) at the top.
- [x] (N/A) New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] (N/A) New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] (N/A) UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/users-teams-organizations/permissions.mdx [hunk 1]
```diff
@@ -235,6 +234,17 @@ Allows members to manage the set of [VCS providers](/cloud-docs/vcs) and [SSH ke
...
 Allows members to publish and delete providers, modules, or both providers and modules in the organization's [private registry](/cloud-docs/registry). These permissions are otherwise only available to organization owners.
 
+### Manage Membership
+
+Allows members to invite users to the organization, remove users from the organization, and add or remove users from teams within the organization.
+
+This permission grants the ability to view the list of users within the organization, and to view the organization access of other visible teams. It does not permit the creation of teams, the ability to modify the settings of existing teams, or the ability to view secret teams.
+
+In order to modify the membership of a team, a user with Manage Membership permissions must have visibility into the team (i.e. the team must be ["Visible"](/cloud-docs/users-teams-organizations/teams#team-visibility), or the user must be on the team).
+In order to remove a user from the organization, the holder of this permission must have visibility into all of the teams which the user is a member of.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/users-teams-organizations/permissions.mdx:244]: Quick nitpick: Should we add a line break in between the above paragraph and this one?
```suggestion

In order to remove a user from the organization, the holder of this permission must have visibility into all of the teams which the user is a member of.
```

File: website/docs/cloud-docs/users-teams-organizations/permissions.mdx [hunk 2]
```diff
@@ -235,6 +234,17 @@ Allows members to manage the set of [VCS providers](/cloud-docs/vcs) and [SSH ke
...
+
+~> This permission is intended to allow owners of large organizations to delegate membership management to another trusted team, and should be granted to only teams of trusted users. **Assign with caution:** Users with this permission are able to add themselves to any visible team, and inherit the permissions of any visible team.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/users-teams-organizations/permissions.mdx:246]: Quick suggestion: Moving the "only" here earlier in the sentence flows a bit better/adds a little more emphasis on the "only" to me -- what do you think?
```suggestion
~> This permission is intended to allow owners of large organizations to delegate membership management to another trusted team, and should only be granted to teams of trusted users. **Assign with caution:** Users with this permission are able to add themselves to any visible team, and inherit the permissions of any visible team.
```

**Review by juliannatetreault — COMMENTED** [review]
I added a couple of suggestions, but this otherwise looks and sounds good to me!

**Review by juliannatetreault — APPROVED** [review]
This looks great! ✨

File: website/docs/cloud-docs/users-teams-organizations/organizations.mdx [hunk 3]
```diff
@@ -107,13 +107,13 @@ Review a list of tags for all resources across the organization. When you delete
 
 All users in an organization can access the **Teams** page, which displays a list of [teams][] within the organiation. This excludes [secret teams](/terraform/cloud-docs/users-teams-organizations/teams#team-visibility) where you are not a member. You can also view team membership and manage team API tokens.
 
-Organization owners can also create and delete teams, manage team membership, and manage team API tokens. Remember that users must accept an invitation to the organization before you can add them to teams.
+Organization owners can also create and delete teams, and manage team API tokens. Both owners, and users with [Manage Membership](/terraform/cloud-docs/users-teams-organizations/permissions#manage-membership) permissions, can manage the membership of teams. Remember that users must accept an invitation to the organization before you can add them to teams.
```
Comment (robin-norwood) [inline website/docs/cloud-docs/users-teams-organizations/organizations.mdx:110]:
```suggestion
Organization owners can also create and delete teams, and manage team API tokens. Both owners and users with [Manage Membership](/terraform/cloud-docs/users-teams-organizations/permissions#manage-membership) permissions can manage the membership of teams. Remember that users must accept an invitation to the organization before you can add them to teams.
```

File: website/docs/cloud-docs/users-teams-organizations/organizations.mdx [hunk 4]
```diff
@@ -107,13 +107,13 @@ Review a list of tags for all resources across the organization. When you delete
...
 
 [permissions-citation]: #intentionally-unused---keep-for-maintainers
 
 #### Users
 
-Organization owners can invite Terraform Cloud users into the organization, cancel invitations, and remove existing members.
+Organization owners, and users with [Manage Membership](/terraform/cloud-docs/users-teams-organizations/permissions#manage-membership) permissions, can invite Terraform Cloud users into the organization, cancel invitations, and remove existing members.
```
Comment (robin-norwood) [inline website/docs/cloud-docs/users-teams-organizations/organizations.mdx:116]:
```suggestion
Organization owners and users with [Manage Membership](/terraform/cloud-docs/users-teams-organizations/permissions#manage-membership) permissions can invite Terraform Cloud users into the organization, cancel invitations, and remove existing members.
```

File: website/docs/cloud-docs/users-teams-organizations/teams.mdx [hunk 5]
```diff
@@ -148,12 +147,12 @@ You can grant teams permissions to manage a project and the workspaces that belo
 
 ## Managing Organization Access
 
+<<<<<<< HEAD
```
Comment (robin-norwood) [inline website/docs/cloud-docs/users-teams-organizations/teams.mdx:150]:
```suggestion
```

File: website/docs/cloud-docs/users-teams-organizations/users.mdx [hunk 6]
```diff
@@ -47,7 +47,7 @@ To use Terraform Cloud or Enterprise, you must create an account through one of
 
 ## Joining Organizations and Teams
 
-An organization owner must [invite you to join their organization](/terraform/cloud-docs/users-teams-organizations/organizations#users). After you join, the organization owner must also [add you to one or more teams](/terraform/cloud-docs/users-teams-organizations/teams#managing-team-membership).
+An organization owner or a user with [Manage Membership](/terraform/cloud-docs/users-teams-organizations/permissions#manage-membership) permissions must [invite you to join their organization](/terraform/cloud-docs/users-teams-organizations/organizations#users), and [add you to one or more teams](/terraform/cloud-docs/users-teams-organizations/teams#managing-team-membership).
```
Comment (robin-norwood) [inline website/docs/cloud-docs/users-teams-organizations/users.mdx:50]:
```suggestion
An organization owner or a user with [Manage Membership](/terraform/cloud-docs/users-teams-organizations/permissions#manage-membership) permissions must [invite you to join their organization](/terraform/cloud-docs/users-teams-organizations/organizations#users) and [add you to one or more teams](/terraform/cloud-docs/users-teams-organizations/teams#managing-team-membership).
```

**Review by robin-norwood — APPROVED** [review]
lgtm, with a few small fixes.

**Pull request merged**